File: src/interfaceAdapters/addTodo.dto.ts
export class AddTodoDto {
  private user_id: number;
  private title: string;

  constructor(user_id: number, title: string) {
    this.user_id = user_id;
    this.title = title;
  }

  getUserId(): number {
    return this.user_id;
  }

  getTitle(): string {
    return this.title;
  }
}

File: src/main/java/com/todoapp/application/dto/AddTodoDto.java
package com.todoapp.application.dto;

/**
 * Todo追加用のDTO
 */
public record AddTodoDto(
    long userId,
    String title
) {
    public AddTodoDto {
        if (userId <= 0) {
            throw new IllegalArgumentException("userId must be positive");
        }
        if (title == null || title.trim().isEmpty()) {
            throw new IllegalArgumentException("title cannot be empty");
        }
        title = title.trim();
    }
} 